Sources/wkhtmltopdf/Document+generatePDF.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+extension Document {
+    public func generatePDF() async throws -> Data {
+        let fileManager = FileManager.default
+
+        let workDir = "/tmp/vapor-wkhtmltopdf"
+        try fileManager.createDirectory(atPath: workDir, withIntermediateDirectories: true)
+
+        defer {
+            try? fileManager.removeItem(atPath: workDir)
+        }
+
+        var wkArgs: [String] = globalArguments()
+
+        let pageFiles: [String] = try pages.enumerated().map { index, page in
+            if page.isUrl, let url = page.url {
+                wkArgs.append(contentsOf: page.toArguments().filter { $0 != url })
+                return url
+            } else {
+                let name = "page_\(index)_\(UUID().uuidString).html"
+                let filename = "\(workDir)/\(name)"
+                try page.content.write(to: URL(fileURLWithPath: filename))
+
+                wkArgs.append(contentsOf: page.toArguments())
+                return filename
+            }
+        }
+
+        wkArgs.append(contentsOf: pageFiles)
+        print("wkhtmltopdf args: \(wkArgs)")
+
+        return try await withCheckedThrowingContinuation { continuation in
+            let wk = Process()
```

Comment:
`Foundation.Process` is terrible. Consider yoinking [the excellent and much less random-failure-prone work Jonathan Grynspan did in SwiftTesting](https://github.com/swiftlang/swift-testing/blob/main/Sources/Testing/ExitTests/SpawnProcess.swift) for the purpose, since `Subprocess` still seems to be a ways off.

Reply:
I hadn't considered it but we could actually also rely on the [Wkhtmltopdf C lib](https://wkhtmltopdf.org/libwkhtmltox/) as a `.systemLibrary` instead

Reply:
Definitely worth a try, integration with a lib is almost always better than dealing with shelling out to processes. (Almost...)